pilot/pkg/bootstrap/server_test.go

```diff
@@ -137,15 +139,12 @@ func TestReloadIstiodCert(t *testing.T) {
 		KeyFile:  keyFile,
 	}
 
-	// setup cert watches.
-	err = s.initCertificateWatches(tlsOptions)
-	for _, fn := range s.startFuncs {
-		if err := fn(stop); err != nil {
-			t.Fatalf("Could not invoke startFuncs: %v", err)
-		}
+	if err = s.server.Start(stop); err != nil {
+		t.Fatalf("Could not invoke startFuncs: %v", err)
 	}
 
-	if err != nil {
+	// setup cert watches.
+	if err = s.initCertificateWatches(tlsOptions); err != nil {
```

Comment:
is this intentionally after server start now?

Reply:
Nope ... restored the original order.